src/lib.rs

```diff
@@ -61,6 +86,19 @@ macro_rules! lsp_enum {
                 }
             }
         }
+
+        impl std::convert::TryFrom<&str> for $typ {
+            type Error = &'static str;
+            fn try_from(value: &str) -> Result<Self, Self::Error> {
+                match value {
+                    $(
+                        _ if { let (buf, len)  = crate::fmt_pascal_case_const(stringify!($name)); &buf[..len] == value.as_bytes() } => Ok(Self::$name),
```

Comment:
great idea. The only weird thing is that when I change the max length from 128 to 5 we will error at runtime (not at compile time!) if the `value` parameter is only known at runtime -- even though `$name` is known at compile time.
Assertions probably won't help here either (though they should be allowed in stable Rust [soon](https://github.com/rust-lang/rust/pull/89508))

So we might save allocations but we still run the code at runtime and don't generate the neat match-statement a human would write, which is `"TypeParameter" => Ok(Self::TYPE_PARAMETER)`

Anyway, that's not so important.

Reply:
So at this point it seems better to write
`match pascal_case_to_screaming_snake_case(value) { "TEXT" => Ok(Self::Text), .. }`